src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1289,4 +1290,1373 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
         this->oneTimeInitFlag = false;
     }
 }
+
+// From here on, the Fuel Fired Heat Pump module EIRFuelFiredHeatPump
+// Enum string definitions
+static constexpr std::array<std::string_view, static_cast<int>(EIRFuelFiredHeatPump::OATempCurveVar::Num)> OATempCurveVarNamesUC = {"DRYBULB",
+                                                                                                                                    "WETBULB"};
+static constexpr std::array<std::string_view, static_cast<int>(EIRFuelFiredHeatPump::WaterTempCurveVar::Num)> WaterTempCurveVarNamesUC = {
+    "ENTERINGCONDENSER", "LEAVINGCONDENSER", "ENTERINGEVAPORATOR", "LEAVINGEVAPORATOR"};
+static constexpr std::array<std::string_view, static_cast<int>(EIRFuelFiredHeatPump::DefrostType::Num)> DefrostTypeNamesUC = {"TIMED", "ONDEMAND"};
+
+void EIRFuelFiredHeatPump::doPhysics(EnergyPlusData &state, Real64 currentLoad)
+{
+    Real64 const reportingInterval = state.dataHVACGlobal->TimeStepSys * DataGlobalConstants::SecInHour;
+
+    // ideally the plant is going to ensure that we don't have a runflag=true when the load is invalid, but
+    // I'm not sure we can count on that so we will do one check here to make sure we don't calculate things badly
+    if ((this->EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredCooling && currentLoad >= 0.0) ||
+        (this->EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredHeating && currentLoad <= 0.0)) {
+        this->resetReportingVariables();
+        return;
+    }
+
+    // get setpoint on the load side outlet
+    Real64 loadSideOutletSetpointTemp = this->getLoadSideOutletSetPointTemp(state);
+
+    // Use a logic similar to that for a boilder: If the specified load is 0.0 or the boiler should not run
+    // then we leave this subroutine. Before leaving
+    // if the component control is SERIESACTIVE we set the component flow to inlet flow so that flow resolver
+    // will not shut down the branch
+    auto &sim_component(DataPlant::CompData::getPlantComponent(state, this->loadSidePlantLoc));
+    bool RunFlag = true;
+    if ((this->EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredHeating && currentLoad <= 0.0) || !RunFlag) {
+        if (sim_component.FlowCtrl == DataBranchAirLoopPlant::ControlType::SeriesActive)
+            this->loadSideMassFlowRate = state.dataLoopNodes->Node(this->loadSideNodes.inlet).MassFlowRate;
+        this->resetReportingVariables();
+        return;
+    }
+
+    DataPlant::PlantLoopData &thisLoadPlantLoop = state.dataPlnt->PlantLoop(this->loadSidePlantLoc.loopNum);
```

Comment:
Can someone do the world a favor and just rename `dataPlnt` to `dataPlant` already? The `data` doesn't bother anyone, but yet the `a` in `Plant` is too much?